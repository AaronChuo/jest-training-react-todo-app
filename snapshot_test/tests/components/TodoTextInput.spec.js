import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TodoTextInput from 'components/TodoTextInput';

describe('<TodoTextInput />', () => {
  let handleSave;
  let wrapper;

  beforeEach(() => {
    handleSave = jest.fn();
    wrapper = shallow(
      <TodoTextInput
        editing
        newTodo
        text=""
        placeholder="Please fill your name"
        onSave={handleSave}
      />
    );
  });

  it('should render initial component', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('simulates Enter key down to save new todo', () => {
    const mockEvent = {
      which: 13,
      target: { value: 'this is a new todo' },
    };
    wrapper.simulate('keyDown', mockEvent);
    expect(handleSave).toHaveBeenCalledWith(mockEvent.target.value);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('simulates Enter key down to save edited todo', () => {
    const mockEvent = {
      which: 13,
      target: { value: 'this is an edited todo' },
    };
    wrapper.setProps({ newTodo: false });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper.simulate('keyDown', mockEvent);
    expect(handleSave).toHaveBeenCalledWith(mockEvent.target.value);
  });

  it('simulates non-Enter key down', () => {
    const mockEvent = {
      which: 19,
      target: { value: '' },
    };
    wrapper.simulate('keyDown', mockEvent);
    expect(handleSave).not.toHaveBeenCalled();
  });

  it('simulates change text', () => {
    const mockEvent = {
      target: { value: 'changed text' },
    };
    wrapper.simulate('change', mockEvent);
    expect(wrapper.state()).toMatchSnapshot();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('simulates blur', () => {
    const mockEvent = {
      target: { value: 'current text'},
    };
    wrapper.simulate('blur', mockEvent);
    expect(handleSave).not.toHaveBeenCalled();
  });

  it('simulates blur to save current text', () => {
    const mockEvent = {
      target: { value: 'current text'},
    };
    wrapper.setProps({ newTodo: false });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    wrapper.simulate('blur', mockEvent);
    expect(handleSave).toHaveBeenCalledWith(mockEvent.target.value);
  });
});
