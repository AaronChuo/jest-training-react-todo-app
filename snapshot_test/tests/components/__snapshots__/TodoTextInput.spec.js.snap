// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoTextInput /> should render initial component 1`] = `
<input
  className="edit new-todo"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Please fill your name"
  type="text"
  value=""
/>
`;

exports[`<TodoTextInput /> simulates Enter key down to save edited todo 1`] = `
<input
  className="edit"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Please fill your name"
  type="text"
  value=""
/>
`;

exports[`<TodoTextInput /> simulates Enter key down to save new todo 1`] = `
<input
  className="edit new-todo"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Please fill your name"
  type="text"
  value=""
/>
`;

exports[`<TodoTextInput /> simulates blur to save current text 1`] = `
<input
  className="edit"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Please fill your name"
  type="text"
  value=""
/>
`;

exports[`<TodoTextInput /> simulates change text 1`] = `
Object {
  "text": "changed text",
}
`;

exports[`<TodoTextInput /> simulates change text 2`] = `
<input
  className="edit new-todo"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Please fill your name"
  type="text"
  value="changed text"
/>
`;
